### PR TITLE
Documentation update with xdp_test

### PR DESCRIPTION
**Pull request by shpalani**
## Description
This PR contains documentation updates related to xdp_test program type. 
It also adds pointers to xdp-for-windows external extension to use for XDP.

## Testing

_Do any existing tests cover this change? N/A Are new tests needed? N/A

## Documentation

_Is there any documentation impact for this change? Yes

## Installation

_Is there any installer impact for this change? No

### CODE REVIEW

File: docs/GettingStarted.md [hunk 1]
```diff
@@ -592,3 +592,7 @@ on (typically the development machine) at driver load time.
 
 This eliminates the need for repeated manual copy of the modified driver binary and saves considerable time during the
 development cycle.  See the [Windows Hardware Developer documentation](https://learn.microsoft.com/en-us/windows-hardware/drivers/debugger/-kdfiles--set-driver-replacement-map-) for more details.
+
+## eBPF external extensions
+Extensions that integrate with eBPF for Windows platform:
```
Comment (dthaler) [inline docs/GettingStarted.md:597]:
```suggestion
Extensions that integrate with eBPF for Windows:
```
grammar

**Review by mtfriesen — APPROVED** [review]

File: docs/InstallEbpf.md [hunk 2]
```diff
@@ -202,3 +202,7 @@ Start an admin Powershell on the Windows Host and run the following command and
     ```cmd
     kubectl apply -f manifests/Kubernetes/ebpf-for-windows-daemonset.yaml
     ```
+
+## Installing eBPF external extensions
+
+- Install [xdp-for-windows](https://github.com/microsoft/xdp-for-windows/releases) extension, if your eBPF program uses XDP driver. Please refer to  [installation here](https://github.com/microsoft/xdp-for-windows/blob/main/docs/usage.md)
```
Comment (dthaler) [inline docs/InstallEbpf.md:208]:
```suggestion
- If your eBPF program requires XDP, install the [xdp-for-windows](https://github.com/microsoft/xdp-for-windows/releases) extension by following the [XDP for Windows installation instructions](https://github.com/microsoft/xdp-for-windows/blob/main/docs/usage.md).
```
grammar and punctuation.

File: docs/tutorial.md [hunk 3]
```diff
@@ -935,12 +935,12 @@ FDs or addresses.  When the program is actually installed, the relocation
 section will be used to insert the actual map address into the executable
 code.
 
-## 5. Installing eBPF programs
+# 5. Installing eBPF programs
 
-`netsh ebpf` can also install your eBPF program. As an example, the program we created above won't do much but we can install it via:
+`netsh ebpf` can also install your eBPF program. As an example, the program we created above won't do much but we can install it with program type `xdp_test` via:
```
Comment (dthaler) [inline docs/tutorial.md:940]:
```suggestion
`netsh ebpf` can also install your eBPF program. As an example, the program we created above won't do much but we can install it using the program type `xdp_test` via:
```

File: docs/tutorial.md [hunk 4]
```diff
@@ -963,7 +963,9 @@ You can also install and interact with the eBPF programs programmatically. See h
 
 Learn more about using eBPF at https://github.com/microsoft/ebpf-for-windows/blob/main/docs/GettingStarted.md#using-ebpf-for-windows
 
-# 7. Next steps
+You can install `xdp-for-windows` extension from https://github.com/microsoft/xdp-for-windows/releases. This [github repository](https://github.com/microsoft/xdp-for-windows) has the Windows native XDP support, with the program type=xdp.
```
Comment (dthaler) [inline docs/tutorial.md:966]:
```suggestion
You can install the `xdp-for-windows` extension from https://github.com/microsoft/xdp-for-windows/releases. Its [github repository](https://github.com/microsoft/xdp-for-windows) has Windows native XDP support, with the program type=xdp.
```
grammar

**Comment by mtfriesen** [comment]
In the long run, it would probably reduce confusion to either entirely remove the `xdp_test` program type from the eBPF repo, or rename it to something to be obviously and unambiguously incompatible with the `xdp` program type. I'd imagine a `sample` program type should be sufficient for sample purposes.

**Comment by shpalani** [comment]
> In the long run, it would probably reduce confusion to either entirely remove the `xdp_test` program type from the eBPF repo, or rename it to something to be obviously and unambiguously incompatible with the `xdp` program type. I'd imagine a `sample` program type should be sufficient for sample purposes.

This tracking issue https://github.com/microsoft/ebpf-for-windows/issues/2974 talks about the removal of xdp_test in the second part.

**Review by saxena-anurag — APPROVED** [review]

**Review by mtfriesen — APPROVED** [review]

File: docs/GettingStarted.md [hunk 5]
```diff
@@ -592,3 +592,7 @@ on (typically the development machine) at driver load time.
 
 This eliminates the need for repeated manual copy of the modified driver binary and saves considerable time during the
 development cycle.  See the [Windows Hardware Developer documentation](https://learn.microsoft.com/en-us/windows-hardware/drivers/debugger/-kdfiles--set-driver-replacement-map-) for more details.
+
+## eBPF external extensions
+Extensions that integrate with eBPF for Windows:
```
Comment (dthaler) [inline docs/GettingStarted.md:597]: @Alan-Jowett Should https://github.com/microsoft/NtosEbpfExt also be in this list?

**Pull request merged**